### PR TITLE
Type GraphQL DateTime scalar in GraphQL Code Generator

### PR DESCRIPTION
**Pull request by Ben-Ho**
needed to improve starter: https://github.com/vivid-planet/comet-starter/pull/10

### CODE REVIEW

**Comment by Ben-Ho** [comment]
@nsams @johnnyomair lint failes but I don't understand why.

who does generate the types used here: https://github.com/vivid-planet/comet/blob/main/demo/admin/src/links/EditLink.tsx#L8, especially GQLUpdateLinkMutation. I would expect them to be based on this gql-query: https://github.com/vivid-planet/comet/blob/main/demo/admin/src/links/EditLink.tsx#L40 or does this generate from somewhere else? (same for savePage)

probably it's because of https://github.com/vivid-planet/comet/blob/main/demo/admin/codegen.ts#L31

lastUpdatedAt in api-schema should be optional, as it's defined as nullable here: https://github.com/vivid-planet/comet/blob/main/demo/api/src/links/links.resolver.ts#L26. Also schema does not contain a !.

**Comment by nsams** [comment]
you are mixing input and response (in response it is non-nullable, in input it is nullable)

the proper fix is `lastUpdatedAt?: string | null;` I guess.

Can you reproduce the problem the pipeline reports locally?

File: packages/admin/cms-admin/src/pages/createUsePage.tsx [hunk 1]
```diff
@@ -52,12 +52,12 @@ interface CreateUsePage {
         GQLUpdatePageMutationVariables extends {
             pageId: string;
             input: BlockGQLData<RootBlocks>;
-            lastUpdatedAt?: string;
+            lastUpdatedAt: string | null;
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/pages/createUsePage.tsx:55]:
```suggestion
            lastUpdatedAt?: string | null;
```

File: packages/admin/cms-admin/src/pages/createUsePage.tsx [hunk 2]
```diff
@@ -52,12 +52,12 @@ interface CreateUsePage {
...
             attachedPageTreeNodeId?: string | null;
         } = {
             pageId: string;
             input: BlockGQLData<RootBlocks>;
-            lastUpdatedAt?: string;
+            lastUpdatedAt: string | null;
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/pages/createUsePage.tsx:60]:
```suggestion
            lastUpdatedAt?: string | null;
```

**Review by johnnyomair — CHANGES_REQUESTED** [review]
You can't remove the optional identifier `?` for `lastUpdatedAt`, as variables in GraphQL mutations can be both absent (= undefined) and set to null. Adding the suggestions I proposed should resolve the pipeline issues

**Comment by johnnyomair** [comment]
I think we should also change the codegen config for the Demo site

**Review by johnnyomair — APPROVED** [review]

**Review by nsams — APPROVED** [review]

**Comment by nsams** [comment]
please continue in starter

**Comment by johnnyomair** [comment]
> please continue in starter

We'll need a stable release for it

**Comment by thomasdax98** [comment]
@johnnyomair @nsams Does this qualify as a patch? Then we could make a 4.4.2 Release. The only thing missing is a changeset for this change

**Comment by nsams** [comment]
Why not, or is there more coming we could wait for?

**Comment by thomasdax98** [comment]
@nsams We could wait for https://github.com/vivid-planet/comet/pull/1266 and https://github.com/vivid-planet/comet/pull/1267

**Pull request merged**